CMakeLists.txt

```diff
@@ -73,6 +73,12 @@ else()
     set(target "${CMAKE_SYSTEM}")
 endif()
 
+# workaround, remove default /W and friends to prevent D9025 warnings
```

Comment:
I talked on the cmake slack channel about this, and its a complicated story, see https://gitlab.kitware.com/cmake/cmake/issues/19084 I commented there as well.
TL;DR; 
cmake adds /W3 by default, and when you also introduce /W4 it warns that both are specified and /W4 will override /W3.

The workaround removes a /Wn flags first to make sure /W4 is the only flag remaining

Reply:
I seem to recall we talked about some other strategies to resolve this, potentially removing all but the last warning flag on the line. Have you tested any other options to prevent these warnings?

Reply:
Alternatively we could just remove all exact matches of /W3 surrounded with either spaces or begin-or-end of word markers.

```
    string (REGEX REPLACE "(^| )[/]W3( |$)" " " CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
    string (REGEX REPLACE "(^| )[/]W3( |$)" " " CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
```

This seems to work well, https://regexr.com/4hrko